synapse/handlers/federation.py

```diff
@@ -1931,7 +1931,7 @@ async def on_make_knock_request(
             builder=builder
         )
 
-        event_allowed = await self.third_party_event_rules.check_event_allowed(
+        event_allowed, _ = await self.third_party_event_rules.check_event_allowed(
```

Comment:
Do we not allow modification of events received over federation?

Reply:
No, to avoid getting splitbrained on what a given event looks like, there's some words on that in the doc.